prio/prg.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2018, Henry Corrigan-Gibbs
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. 
+ */
+
+#include <mprio.h>
+#include <nss/pk11pub.h>
+#include <strings.h>
+
+#include "prg.h"
+#include "rand.h"
+#include "share.h"
+#include "util.h"
+
+#define AES_BLOCK_SIZE 16
+
+struct prg {
+  PK11SlotInfo *slot;
+  PK11SymKey *key;
+  PK11Context *ctx;
+};
+
+SECStatus 
+PrioPRGSeed_randomize (PrioPRGSeed *key)
+{
+  return rand_bytes ((unsigned char *)key, PRG_SEED_LENGTH);   
+}
+
+PRG 
+PRG_new (const PrioPRGSeed key_in)
+{
+  PRG prg = malloc (sizeof (*prg));
```

Comment:
`sizeof(PRG)`?

Reply:
These are equivalent, right? To make the code easier to read, I'll change to `sizeof(PRG)`.

Reply:
Whoops. After running valgrind, I realized that this change introduced a bug.

The `sizeof(PRG)` is the size of a `struct prg*` (i.e., a pointer) while `sizeof(*prg)` is the size of a `struct prg` (i.e., the struct itself). The latter one is what I wanted. To make this clear, I will change the code to say `sizeof(struct prg)`.